demo/site/src/middleware/redirectToMainHost.ts

```diff
@@ -1,8 +1,25 @@
+import type { PublicSiteConfig } from "@src/site-configs";
 import { getHostByHeaders, getSiteConfigForHost, getSiteConfigs } from "@src/util/siteConfig";
 import { NextRequest, NextResponse } from "next/server";
 
 import { CustomMiddleware } from "./chain";
 
+const normalizeDomain = (host: string) => (host.startsWith("www.") ? host.substring(4) : host);
+
+const matchesHostWithAdditionalDomain = (siteConfig: PublicSiteConfig, host: string) => {
+    if (normalizeDomain(siteConfig.domains.main) === normalizeDomain(host)) return true; // non-www redirect
```

Comment:
Is this the correct place to perform the non-www redirect? Shouldn't this be done in the CDN/ingress/whatever (I don't know where exactly we did this before we had site configs).